pandas/tests/dtypes/test_inference.py

```diff
@@ -66,20 +66,39 @@ def __getitem__(self):
 @pytest.mark.parametrize(
     "ll",
     [
-        [], [1], (1, ), (1, 2), {'a': 1},
-        {1, 'a'}, Series([1]),
-        Series([]), Series(['a']).str,
-        np.array([2])])
+        [], [1], tuple(), (1, ), (1, 2), {'a': 1}, {1, 'a'}, np.array([2]),
+        Series([1]), Series([]), Series(['a']).str, Index([]), Index([1]),
+        DataFrame(), DataFrame([[1]]), iter([1, 2]), (x for x in [1, 2]),
+        np.ndarray((2,) * 2), np.ndarray((2,) * 3), np.ndarray((2,) * 4)
+    ])
 def test_is_list_like_passes(ll):
     assert inference.is_list_like(ll)
 
 
-@pytest.mark.parametrize(
-    "ll", [1, '2', object(), str, np.array(2)])
+@pytest.mark.parametrize("ll", [1, '2', object(), str, np.array(2)])
 def test_is_list_like_fails(ll):
     assert not inference.is_list_like(ll)
 
 
+@pytest.mark.parametrize(
```

Comment:
I would prefer to have 2 tests total to avoid the duplication of the args here (IOW 1 for allow_sets=True and 1 for False).

Reply:
not sure if my solution is what you had in mind, but I gave it a shot

Reply:
I didn't see the earlier version, but I don't think this is what Jeff had in mind. If we want to de-duplicate the arguments, you would need a fixture giving them

```python
@pytest.fixture(params=...)
def maybe_list_like(request):
    return request.param
```

Each of the params would be a tuple like `([], True)`, `('2', False)`, and I guess something like `({}, None)` or `({}, 'maybe'})` for set-likes.

Then we would have two tests. In the first we do

```python
obj, expected = ...
if expected:
    expected = True

assert is_list_like(obj) is expected
```


and in the second

```python
if expected is None:
    expected = False

assert is_list_like(obj, include_sets=False) is expected
```

Reply:
yeah @TomAugspurger suggestion is good here. The issues is we can't list the args twice.